runtime/src/bank.rs

```diff
@@ -331,33 +331,49 @@ impl CachedExecutors {
         })
     }
 
-    fn put(&mut self, pubkey: &Pubkey, executor: Arc<dyn Executor>) {
-        let entry = if let Some(mut entry) = self.executors.remove(pubkey) {
-            entry.executor = executor;
-            entry
-        } else {
+    fn put(&mut self, executors: &[(&Pubkey, Arc<dyn Executor>)]) {
+        let new_executors: Option<Vec<_>> = executors
+            .iter()
+            .map(|(key, executor)| {
+                if let Some(mut entry) = self.executors.remove(key) {
+                    entry.executor = executor.clone();
+                    let _ = self.executors.insert(**key, entry);
+                    None
+                } else {
+                    Some((key, executor))
+                }
+            })
+            .collect();
+
+        if let Some(mut new_executors) = new_executors {
             if self.executors.len() >= self.max {
-                let mut least = u64::MAX;
-                let default_key = Pubkey::default();
-                let mut least_key = &default_key;
-
-                for (key, entry) in self.executors.iter() {
-                    let count = entry.prev_epoch_count + entry.epoch_count.load(Relaxed);
-                    if count < least {
-                        least = count;
-                        least_key = key;
+                for _ in 0..new_executors.len() {
```

Comment:
Once I push these changes to the primer count randomization, we'll have a sorted `Vec<(&Pubkey, u64`, here.  With that we can just `take()` the bottom `new_executors.len()` instead of doing this `O(M*N)` thing.

Reply:
These are up now @ #22333